sdk/servicebus/Azure.Messaging.ServiceBus/src/RuleManager/ServiceBusRuleManager.cs

```diff
@@ -0,0 +1,256 @@
+        {
+            Argument.AssertNotClosed(IsDisposed, nameof(ServiceBusRuleManager));
+            Argument.AssertNotNull(description, nameof(description));
+            cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+            EntityNameFormatter.CheckValidRuleName(description.Name);
+            ServiceBusEventSource.Log.AddRuleStart(Identifier, description.Name);
+
+            try
+            {
+                await InnerRuleManager.AddRuleAsync(
+                    description,
+                    cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception exception)
+            {
+                ServiceBusEventSource.Log.AddRuleException(Identifier, exception);
+                throw;
+            }
+
+            cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+            ServiceBusEventSource.Log.AddRuleComplete(Identifier);
+        }
+
+        /// <summary>
+        /// Removes the rule on the subscription identified by <paramref name="ruleName" />.
+        /// </summary>
+        ///
+        /// <param name="ruleName">Name of the rule</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        ///
+        /// <returns>A task instance that represents the asynchronous remove rule operation.</returns>
+        public virtual async Task RemoveRuleAsync(
+            string ruleName,
+            CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotClosed(IsDisposed, nameof(ServiceBusRuleManager));
+            Argument.AssertNotNullOrEmpty(ruleName, nameof(ruleName));
+            cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+            ServiceBusEventSource.Log.RemoveRuleStart(Identifier, ruleName);
+
+            try
+            {
+                await InnerRuleManager.RemoveRuleAsync(
+                    ruleName,
+                    cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception exception)
+            {
+                ServiceBusEventSource.Log.RemoveRuleException(Identifier, exception);
+                throw;
+            }
+
+            cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+            ServiceBusEventSource.Log.RemoveRuleComplete(Identifier);
+        }
+
+        /// <summary>
+        /// Get all rules associated with the subscription.
+        /// </summary>
+        ///
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        ///
+        /// <returns>Returns a list of rules description</returns>
+        public virtual async Task<IEnumerable<RuleDescription>> GetRulesAsync(CancellationToken cancellationToken = default)
```

Comment:
This should probably return an IList rather than an IEnumerable as per [guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-service-return-model-collections).